### PR TITLE
fix: puts navbar functionality behind feature flag

### LINKED ISSUE #24515: [Bug]: Make showNavbar in Embed view internal feature
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

The embed view with navbar should not hide share features.

### Steps To Reproduce

1. Open any app
2. Go to app settings
3. In the `Share and embed` settings, click `show Navabar`
4. Open the embedded view

### Public Sample App

_No response_

### Environment

Production

### Issue video log

_No response_

### Version

Appsmith v1.9.22-SNAPSHOT

### PR DESCRIPTION
**Pull request by rahulbarwal**
## Description
This PR puts shownavbar functionality behind a feature flag.

* adds analytics events to track apps that use this flag.
* updates relevant specs.

#### PR fixes following issue(s)
Fixes #24515

#### Type of change

- Bug fix (non-breaking change which fixes an issue)
#### How Has This Been Tested?
- [x] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Comment by rahulbarwal** [comment]
Hey @shwetha-ramesh, 
This needs to be tested from 2 user accounts:
1. Appsmith mail id
2. Any other mail ID

DP URL : https://ce-24561.dp.appsmith.com/
Steps: 
1. open any app
2. Go to app settings
3. Open share and embed settings
4. Make application public(if not already)
5. toggle `show navigation bar`
      a. Appsmith mail id : it should add and remove 'navbar=true` from snippet URL
                1. when show nav bar is set : https://ce-24561.dp.appsmith.com/app/my-first-application/page1-648c29c5ac2df1623cf16dbc?embed=true&navbar=true
                2. when show nav bar is unset : https://ce-24561.dp.appsmith.com/app/my-first-application/page1-648c29c5ac2df1623cf16dbc?embed=true
      b. Any other mail id : it should add and remove 'embed=true` from snippet URL
                1. when show nav bar is set : https://ce-24561.dp.appsmith.com/app/my-first-application/page1-648c29c5ac2df1623cf16dbc
                2. when show nav bar is unset : https://ce-24561.dp.appsmith.com/app/my-first-application/page1-648c29c5ac2df1623cf16dbc?embed=true

File: app/client/src/pages/AppViewer/Navigation/index.tsx [hunk 1]
```diff
@@ -60,6 +62,12 @@ export function Navigation() {
       ?.orientation,
   ]);
 
+  if (showNavBar) {
+    AnalyticsUtil.logEvent("APP_VIEW_WITH_NAVBAR_FLAG", {
+      appUrl: currentApplicationDetails?.id,
+    });
+  }
```
Comment (akash-codemonk) [inline app/client/src/pages/AppViewer/Navigation/index.tsx:69]: Inside a useEffect maybe? This will get called on every render.

Comment (rahulbarwal) [reply app/client/src/pages/AppViewer/Navigation/index.tsx]: Makes sense. added.

**Review by Aishwarya-U-R — CHANGES_REQUESTED** [review]
Hey @rahulbarwal - [both the new cases are flaky](https://cloud.cypress.io/projects/eyxvp8/runs/36343/test-results?actions=%5B%5D&browsers=%5B%5D&groups=%5B%5D&isFlaky=%5B%7B%22value%22%3Atrue%2C%22label%22%3A%22Flaky%22%7D%5D&modificationDateRange=%7B%22startDate%22%3A%221970-01-01%22%2C%22endDate%22%3A%222038-01-19%22%7D&orderBy=EXECUTION_ORDER&oses=%5B%5D&specs=%5B%5D&statuses=%5B%5D&testingTypesEnum=%5B%5D). Please fix the flakyness & then merge the PR. NO merging till then.

**Comment by shwetha-ramesh** [comment]
@rahulbarwal QA Done.

File: app/client/cypress/support/Pages/AppSettings/EmbedSettings.ts [hunk 2]
```diff
@@ -63,4 +65,20 @@ export class EmbedSettings {
...
   }
+
+  public TogglePublicAccess(check: true | false = true) {
+    const input = this.agHelper.GetElement(
+      this.locators._enablePublicAccessSettingsPage,
+    );
+    input.invoke("attr", "checked").then((value) => {
+      if (value !== check.toString()) {
+        this.agHelper.GetNClick(this.locators._enablePublicAccessSettingsPage);
+        cy.wait("@changeAccess").should(
```
[the remark this answers is not in the document]
Comment (rahulbarwal) [reply app/client/cypress/support/Pages/AppSettings/EmbedSettings.ts]: Done.

File: app/client/cypress/e2e/Regression/ClientSide/SettingsPane/EmbedSettings_spec.ts [hunk 3]
```diff
@@ -79,18 +79,46 @@ describe("In-app embed settings", () => {
     ValidateSyncWithInviteModal("false");
   });
 
-  it("5. Changing the show navigation bar setting in the App settings pane should update the embed URL", () => {
+  it("5. [Feature flag APP_EMBED_VIEW_HIDE_SHARE_SETTINGS_VISIBILITY=false] Changing the show navigation bar setting in the App settings pane should update the embed URL", () => {
+    cy.intercept("GET", "/api/v1/users/features", {
+      fixture: "featureFlags.json",
+    }).as("featureFlags");
     _.agHelper.RefreshPage();
```
[the remark this answers is not in the document]
Comment (rahulbarwal) [reply app/client/cypress/e2e/Regression/ClientSide/SettingsPane/EmbedSettings_spec.ts]: Done.

File: app/client/src/utils/AnalyticsUtil.tsx [hunk 4]
```diff
@@ -58,6 +58,7 @@ export type EventName =
   | "SIDEBAR_NAVIGATION"
   | "PUBLISH_APP"
   | "PREVIEW_APP"
+  | "APP_VIEW_WITH_NAVBAR_FLAG"
```
Comment (dhruvikn) [inline app/client/src/utils/AnalyticsUtil.tsx:61]: Just a minor nitpick, can we please rename this event to "APP_VIEWED_WITH_NAVBAR"?

**Comment by dhruvikn** [comment]
The rest of the PR looks good to me @rahulbarwal.

File: app/client/src/utils/AnalyticsUtil.tsx (hunk 4 above)
Comment (rahulbarwal) [reply app/client/src/utils/AnalyticsUtil.tsx]: Done.

**Review by dhruvikn — APPROVED** [review]

File: app/client/cypress/e2e/Regression/ClientSide/SettingsPane/EmbedSettings_spec.ts [hunk 5]
```diff
@@ -87,6 +87,7 @@ describe("In-app embed settings", () => {
     agHelper.RefreshPage();
     embedSettings.OpenEmbedSettings();
     embedSettings.TogglePublicAccess(true);
+    agHelper.Sleep();
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/SettingsPane/EmbedSettings_spec.ts:90]: If making an app public toggle - will always need some wait - can we consider adding this into TogglePublicAccess() itself? So any future use of the method is also taken care.

Comment (rahulbarwal) [reply app/client/cypress/e2e/Regression/ClientSide/SettingsPane/EmbedSettings_spec.ts:90]: @Aishwarya-U-R I was testing a theory with this timeout; I was wrong. This Sleep would not be needed. CI run still marked these tests flaky. 
On local everything passes through in 1st run. Debugging this.

Comment (Aishwarya-U-R) [reply app/client/cypress/e2e/Regression/ClientSide/SettingsPane/EmbedSettings_spec.ts]: Got it, please do!

**Review by Aishwarya-U-R — COMMENTED** [review]
Cypress looks good, what action did we exactly take here for the flakiness on new tests please?

**Comment by rahulbarwal** [comment]
> Cypress looks good, what action did we exactly take here for the flakiness on new tests please?
@Aishwarya-U-R I had copied code to invoke the `checked` attribute from some location, but it was not evaluating correct value due to which `if` condition was failing.

Updated my function to invoke `prop` instead of `attr` and the `if` condition evaluated correctly.https://github.com/appsmithorg/appsmith/pull/24561/files#diff-0f3bbf2d7e8fd1cfcde4bef9e37b60194290ea0dc01b09135590c40f18542da9R73

------
This probably holds true for other locations as well where we are invoking `attr` to check "any" checkbox. They were outside the purview of this PR hence did not check them. 

This will serve as a learning to us as well. How I verified this?
Simply call the toggle function twice with same parameter. Since it must have checked the input the first time it should not go inside the `if` condition second time.

**Comment by Aishwarya-U-R** [comment]
> Updated my function to invoke `prop` instead of `attr` and the `if` condition evaluated correctly.https://github.com/appsmithorg/appsmith/pull/24561/files#diff-0f3bbf2d7e8fd1cfcde4bef9e37b60194290ea0dc01b09135590c40f18542da9R73

Nice & Noted! 🙌🏻

**Review by Aishwarya-U-R — COMMENTED** [review]
Approving Cypress changes!

**Review by akash-codemonk — DISMISSED** [review]

**Review by dhruvikn — DISMISSED** [review]

**Comment by Aishwarya-U-R** [comment]
@rahulbarwal - [Fork_Template_Existing_app_spec.js has failed](https://cloud.cypress.io/projects/eyxvp8/runs/36529/test-results?actions=%5B%5D&browsers=%5B%5D&groups=%5B%5D&isFlaky=%5B%5D&modificationDateRange=%7B%22startDate%22%3A%221970-01-01%22%2C%22endDate%22%3A%222038-01-19%22%7D&orderBy=EXECUTION_ORDER&oses=%5B%5D&specs=%5B%5D&statuses=%5B%7B%22value%22%3A%22FAILED%22%2C%22label%22%3A%22FAILED%22%7D%5D&testingTypesEnum=%5B%5D) & re-run is not working properly due to some issue we are debugging. Can you please confirm if this test is a pass in this branch local run?

**Pull request merged**